go/vt/vtgate/engine/route.go

```diff
@@ -1122,15 +1007,15 @@ func (route *Route) execShard(vcursor VCursor, query string, bindVars map[string
 	}, isDML, canAutocommit)
 }
 
-func (route *Route) anyShard(vcursor VCursor, keyspace *vindexes.Keyspace) (string, string, error) {
+func anyShard(vcursor VCursor, keyspace *vindexes.Keyspace) (string, string, error) {
 	ks, allShards, err := vcursor.GetKeyspaceShards(keyspace)
 	if err != nil {
 		return "", "", err
 	}
 	return ks, allShards[0].Name, nil
 }
 
-func (route *Route) getShardQueries(query string, params *scatterParams) map[string]*querypb.BoundQuery {
+func getShardQueries(query string, params *scatterParams) map[string]*querypb.BoundQuery {
```

Comment:
This was because Route was the only primitive that would send queries. Now, any of the four can send, which requires this to be a stand-alone function.